lib/backend/k8s/resources/systemnetworkpolicies_conversion.go

```
@@ -0,0 +1,75 @@
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/thirdparty"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/converter"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ThirdPartyToSystemNetworkPolicy takes a Kubernetes ThirdPartyResource representation
+// of a Calico Policy and returns the equivalent SystemNetworkPolicy object.
+func ThirdPartyToSystemNetworkPolicy(t *thirdparty.SystemNetworkPolicy) *model.KVPair {
+
+	// Since we are using the Calico API Spec definition to store the Calico
+	// Policy, use the client conversion helper to convert between KV and API.
+	policyName := fmt.Sprintf("%s%s", SystemNetworkPolicyNamePrefix, t.Metadata.Name)
+	r := api.Policy{
+		Metadata: api.PolicyMetadata{
+			Name: policyName,
+		},
+		Spec: t.Spec,
+	}
+	kvp, _ := converter.PolicyConverter{}.ConvertAPIToKVPair(r)
+	kvp.Revision = t.Metadata.ResourceVersion
+
+	return kvp
+}
+
+// SystemNetworkPolicyToThirdParty takes a Calico Policy and returns the equivalent
+// ThirdPartyResource representation.
+func SystemNetworkPolicyToThirdParty(kvp *model.KVPair) *thirdparty.SystemNetworkPolicy {
+	r, _ := converter.PolicyConverter{}.ConvertKVPairToAPI(kvp)
+
+	tprName := systemNetworkPolicyTprName(kvp.Key)
+	tpr := thirdparty.SystemNetworkPolicy{
+		Metadata: metav1.ObjectMeta{
+			Name: tprName,
+		},
+		Spec: r.(*api.Policy).Spec,
+	}
+	if kvp.Revision != nil {
+		tpr.Metadata.ResourceVersion = kvp.Revision.(string)
+	}
+	return &tpr
+}
+
+// systemNetworkPolicyTprName converts a Policy (specifically a System Network Policy)
+// name to a TPR name.
+func systemNetworkPolicyTprName(key model.Key) string {
+	// The name should be policed before we get here.
+	pk := key.(model.PolicyKey)
+	if !strings.HasPrefix(pk.Name, SystemNetworkPolicyNamePrefix) {
+		panic("System Network Policy name is not correctly namespaced")
```

why panic instead of log.Fatal() or something else?